### PR TITLE
Skip asset precomp on non-preview heroku envs

### PR DESCRIPTION
**Pull request by mike9005**
Following a nice suggestion from PK [here](https://github.com/blueapron/blueapron.com/pull/7941#issuecomment-299336184) on skipping redoing asset precompilation on staging and prod heroku environments without needing to add dummy manifest files, and still run precompilation for preview environments.

@blueapron/devops @blueapron/consumer

### CODE REVIEW

**Comment by designblooz** [comment]
Should reserve some herb server and configure it to use this branch to build and deploy

File: lib/language_pack/rails4.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def default_assets_cache
   def run_assets_precompile_rake_task
     instrument "rails4.run_assets_precompile_rake_task" do
       log("assets_precompile") do
-        if Dir.glob('public/assets/manifest-*.json').any?
+        if rake_env["RAILS_ENV"] != "preview" || Dir.glob('public/assets/manifest-*.json').any?
```
Comment (PHPCEO) [inline lib/language_pack/rails4.rb]: How about using `unless rake_env["RAILS_ENV"] == "preview" ...` ?

Comment (HoyaBoya) [inline lib/language_pack/rails4.rb]: Seems to make sense to me:

https://github.com/bbatsov/ruby-style-guide#unless-for-negatives

Comment (travisofthenorth) [inline lib/language_pack/rails4.rb]: `unless` with two conditions is a bit harder to parse. I would leave it as is.

**Comment by mike9005** [comment]
The move to AWS is so close to happening, I'm just abandoning this entirely

**Pull request closed without merging**